pr_agent/git_providers/bitbucket_server_provider.py

```diff
@@ -0,0 +1,353 @@
+                    original_file_content_str = ""
+                case 'DELETE':
+                    edit_type = EDIT_TYPE.DELETED
+                    new_file_content_str = ""
+                    original_file_content_str = self.get_file(file_path, base_sha)
+                    if isinstance(original_file_content_str, (bytes, bytearray)):
+                        original_file_content_str = original_file_content_str.decode("utf-8")
+                case 'RENAME':
+                    edit_type = EDIT_TYPE.RENAMED
+                case _:
+                    edit_type = EDIT_TYPE.MODIFIED
+                    original_file_content_str = self.get_file(file_path, base_sha)
+                    if isinstance(original_file_content_str, (bytes, bytearray)):
+                        original_file_content_str = original_file_content_str.decode("utf-8")
+                    new_file_content_str = self.get_file(file_path, head_sha)
+                    if isinstance(new_file_content_str, (bytes, bytearray)):
+                        new_file_content_str = new_file_content_str.decode("utf-8")
+
+            patch = load_large_diff(file_path, new_file_content_str, original_file_content_str)
+
+            diff_files.append(
+                FilePatchInfo(
+                    original_file_content_str,
+                    new_file_content_str,
+                    patch,
+                    file_path,
+                    edit_type=edit_type,
+                )
+            )
+
+        self.diff_files = diff_files
+        return diff_files
+
+    def publish_comment(self, pr_comment: str, is_temporary: bool = False):
+        if not is_temporary:
+            self.bitbucket_client.add_pull_request_comment(self.workspace_slug, self.repo_slug, self.pr_num, pr_comment)
+
+    def remove_initial_comment(self):
+        try:
+            for comment in self.temp_comments:
+                self.remove_comment(comment)
+        except ValueError as e:
+            get_logger().exception(f"Failed to remove temp comments, error: {e}")
+
+    def remove_comment(self, comment):
+        pass
+
+    # funtion to create_inline_comment
+    def create_inline_comment(self, body: str, relevant_file: str, relevant_line_in_file: str):
+        position, absolute_position = find_line_number_of_relevant_line_in_file(
+            self.get_diff_files(),
+            relevant_file.strip('`'),
+            relevant_line_in_file
+        )
+        if position == -1:
+            if get_settings().config.verbosity_level >= 2:
+                get_logger().info(f"Could not find position for {relevant_file} {relevant_line_in_file}")
+            subject_type = "FILE"
+        else:
+            subject_type = "LINE"
+        path = relevant_file.strip()
+        return dict(body=body, path=path, position=absolute_position) if subject_type == "LINE" else {}
+
+    def publish_inline_comment(self, comment: str, from_line: int, file: str):
```

Comment:
**Suggestion:** Add type hints to the function parameters of `BitbucketServerProvider.publish_inline_comment()`.
```suggestion
    def publish_inline_comment(
      self, comment: str, from_line: int, file: str
    ) -> requests.Response:
```